src/shared/parameter_v2/config/examples/example_config.c

```diff
@@ -0,0 +1,93 @@
+#include "shared/parameter_v2/config/examples/example_config.h"
+
+#include <memory.h>
+#include <stdlib.h>
+
+#include "shared/parameter_v2/config/examples/parameter.h"
+
+// this function should get called in the main.c
+const ThunderbotsConfig_t* app_dynamic_parameters_create(void)
+{
+    // Allocate memory for all configs
+    ThunderbotsConfig_t* thunderbots_config =
+        (ThunderbotsConfig_t*)malloc(sizeof(ThunderbotsConfig_t));
+    ExampleConfig_t* example_config = (ExampleConfig_t*)malloc(sizeof(ExampleConfig_t));
+    FooConfig_t* foo_config         = (FooConfig_t*)malloc(sizeof(FooConfig_t));
+    BarConfig_t* bar_config         = (BarConfig_t*)malloc(sizeof(BarConfig_t));
+
+    // Create "initialization structs"
+    ExampleConfig_t example_config_init = {
+        .FooConfig           = foo_config,
+        .BarConfig           = bar_config,
+        .example_bool_param  = app_dynamic_parameters_createBoolParameter(false),
+        .example_uint_param  = app_dynamic_parameters_createUIntParameter(3),
+        .example_int_param   = app_dynamic_parameters_createIntParameter(3),
+        .example_float_param = app_dynamic_parameters_createFloatParameter(4.04f),
+        .example_string_param =
+            app_dynamic_parameters_createStringParameter("Hello World")};
+
+    BarConfig_t bar_config_init = {
+        .bar_int  = app_dynamic_parameters_createIntParameter(3),
+        .bar_bool = app_dynamic_parameters_createBoolParameter(false),
+    };
+
+    FooConfig_t foo_config_init = {
+        .foo_int  = app_dynamic_parameters_createIntParameter(3),
+        .foo_bool = app_dynamic_parameters_createBoolParameter(false),
+    };
+
+    // Initialize Configs
+    //
+    // Memcpy allows us to NOT hit any undefined behaviour when initializing a const param
+    // after malloc.
+    //
+    // We copy all the values from the stack-allocated struct into the
+    // heap-malloc'd struct.
+    //
+    // https://stackoverflow.com/questions/9691404/how-to-initialize-const-in-a-struct-in-c-with-malloc
+    memcpy(example_config, &example_config_init, sizeof(ExampleConfig_t));
+    memcpy(bar_config, &bar_config_init, sizeof(BarConfig_t));
+    memcpy(foo_config, &foo_config_init, sizeof(FooConfig_t));
+
+    // Resolve Includes
+    //
+    // We "link" the configs together when requested w/ an include statement in the yaml
+    //
+    // include:
+    //      - "foo.yaml"
+    //      - "bar.yaml"
+    thunderbots_config->ExampleConfig = example_config;
+    example_config->FooConfig         = foo_config;
+    example_config->BarConfig         = bar_config;
+
+    return thunderbots_config;
+}
+
+// autogenerated "destructor"
+void app_dynamic_parameters_destroy(const ThunderbotsConfig_t* tbots_config)
+{
+    app_dynamic_parameters_destroyIntParameter(
+        tbots_config->ExampleConfig->FooConfig->foo_int);
```

Comment:
can we do a recursive destructor?

Reply:
yeah we could if we stored the ptrs to the `configs` in an array, but this file is autogenerated, so we can _generate_ recursively and destroy "sequentially" in the correct order